packages/runtime/container-runtime/src/channelCollection.ts

```diff
@@ -120,10 +120,22 @@ interface FluidDataStoreMessage {
 	type: string;
 }
 
+/**
+ * This version of the interface is private to this the package. it should never be exported under any tag.
+ * It is used to manage interactions between layers within the package itself. If something is needed
+ * cross package, it is likely it is also being used cross layer (ContainerRuntime * DataStoreRuntime).
+ * If that is the case, the change likely needs to be staged directly on IFluidParentContext. Changes
+ * being staged on IFluidParentContext can be added here as well, likely with optionality removed,
+ * to ease interactions between layers within this package.
+ */
+export interface IFluidParentContextPrivate extends Omit<IFluidParentContext, "isReadOnly"> {
+	readonly isReadOnly: () => boolean;
+}
+
 /**
  * Creates a shallow wrapper of {@link IFluidParentContext}. The wrapper can then have its methods overwritten as needed
  */
-export function wrapContext(context: IFluidParentContext): IFluidParentContext {
+export function wrapContext(context: IFluidParentContextPrivate): IFluidParentContextPrivate {
 	const isReadOnly = context.isReadOnly;
```

Comment:
nit: inline this now

Reply:
the method says it `Creates a shallow wrapper`, so i'm not sure we should just expose. maybe the comment is wrong, but the current behavior seems most inline with it.

Reply:
I just mean do `isReadonly: () => context.isReadonly()` below.  Anyway, doesn't matter.